docs/source/01_introduction/01_tutorial.md

```diff
@@ -0,0 +1,131 @@
+# Tutorial
+
+
+## Introduction
+
+## Pre-requisite
+
+### Create a virtual environment
+
+It is recommended to create a virutal environment to install your packages. You can install ``kedro-pandera`` from PyPI.
+```bash
+conda create -n kedro-pandera-tutorial python=3.10 -y
+conda activate kedro-pandera-tutorial
+pip install kedro-pandera
+```
+
+Notice that installing ``kedro-pandera`` will automatically install ``kedro``.
+
+```{note}
+``kedro-pandera`` only works with ``kedro>=0.18.13``. You may need to upgrade your kedro version if you are using it in an older project.
+```
+
+### Create the project
+
+We will use the ``pandas-iris`` [kedro starter](https://github.com/kedro-org/kedro-starters/tree/main/pandas-iris) in this tutorial. First, we create a project from this starter:
+
+```bash
+kedro new --starter=pandas-iris
+```
+
+Press enter for each question to have default settings.
+
+### Configure the project
+
+if you are using  ``kedro>=0.18.13, <0.19``, you need to configure your ``settings.py`` to use the ``OmegaConfigLoader``: which is the only ``ConfigLoader``  compatible with ``kedro-pandera``.
+
+```python
+# settings.py
+from kedro.config import OmegaConfigLoader
+
+CONFIG_LOADER_CLASS = OmegaConfigLoader
+```
+```
+
+```{note}
+In ``kedro>=0.19``, ``OmegaConfigLoader`` will be the default and this will step no longer be necessary.
+```
+## Validating data at runtime
+
+### Goal of the tutorial
+
+The goal of ``kedro-pandera`` is to validate data at runtime to ensure they meet some requirements you have previously defined (column names, types, value range...).
+
+```{warning}
+This means that data will be validated each time you run ``kedro run`` command. This may come with some performance penalty.
+```
+
+In our example the pipeline will "train an example machine learning pipeline that runs a 1-nearest neighbour classifier to classify an iris" according to the documentation.
+
+We want to add checks make sure that the input dataset will not vary too much during future retraining on new data. Hence, we will validate ``example_iris_data``  with pandera.
+
+
+### Step 1: Generate the schema
+
+Writing a configuration file for validation is long and not easy to do manually, so we will generate a default schema as a baseline witht the ``kedro-pandera`` CLI ``infer`` command:
+
+```bash
+kedro pandera infer -d example_iris_data
+```
+You should see the following message:
+
+```bash
+[08/09/23 15:24:01] INFO     Loading data from 'example_iris_data' (CSVDataSet)...                                                  data_catalog.py:475
+The schema is written to /path/to/project/pandas-iris/conf/base/catalog_example_iris_data.yml
+```
+
+This will create a file "/path/to/project/pandas-iris/conf/base/catalog_example_iris_data.yml" with the default configuration.
+
+### Step 2: Update the catalog
+
+```yaml
+example_iris_data:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/iris.csv
+  metadata:
+    pandera:
```

Comment:
I've hesitated a lot about this, and don't have strong feelings. 

"kedro-pandera", "kedro_pandera" or "pandera"? Last one if faster to write and I am lazy 😄, but maybe a more explicit key would avoid conflict is someone decided to use this key too. I Guess we should use "kedro_pandera" (with an underscore)  rather than "kedro-pandera" (with a dash) to avoid any potential non valid python name if the dict was converted into a object in the future though.  

Happy to change it. Maybe @datajoely have some thoughts here too?

Reply:
I think it's fine like this :)

Reply:
@Galileo-Galilei Good point. I am happy with both `kedro_pandera` and `pandera`. I have bias due to https://github.com/kedro-org/kedro-viz/blob/6e1d127fb85349ca7244e76719b756cc5001784f/package/kedro_viz/data_access/repositories/catalog.py#L53-L68.

Reply:
Let's stay with ``pandera``, ``kedro_pandera`` with an underscore would not really add consistency since ``kedro-viz`` is written with a dash.